es77x/src/main/scala/tech/beshu/ror/es/request/context/types/GetEsRequestContext.scala

```diff
@@ -0,0 +1,145 @@
+      val indices = indicesOrWildcard(indicesFrom(actionRequest))
+      logger.debug(s"[${id.show}] Discovered indices: ${indices.map(_.show).mkString(",")}")
+      indices
+    },
+    None
+  )
+
+  override def modifyWhenIndexNotFound: ModificationResult = {
+    if (aclContext.doesRequirePassword) {
+      val nonExistentIndex = initialBlockContext.randomNonexistentIndex()
+      if (nonExistentIndex.hasWildcard) {
+        val nonExistingIndices = NonEmptyList
+          .fromList(initialBlockContext.nonExistingIndicesFromInitialIndices().toList)
+          .getOrElse(NonEmptyList.of(nonExistentIndex))
+        update(actionRequest, nonExistingIndices, initialBlockContext.filter)
+        Modified
+      } else {
+        ShouldBeInterrupted
+      }
+    } else {
+      update(actionRequest, NonEmptyList.of(initialBlockContext.randomNonexistentIndex()), initialBlockContext.filter)
+      Modified
+    }
+  }
+
+  override protected def modifyRequest(blockContext: GetEsRequestBlockContext): ModificationResult = {
+    NonEmptyList.fromList(blockContext.indices.toList) match {
+      case Some(indices) =>
+        update(actionRequest, indices, blockContext.filter)
+      case None =>
+        logger.warn(s"[${id.show}] empty list of indices produced, so we have to interrupt the request processing")
+        ShouldBeInterrupted
+    }
+  }
+
+  private def indicesFrom(request: GetRequest): Set[IndexName] = {
+    val indexName = IndexName
+      .fromString(request.index())
+      .getOrElse {
+        throw RequestSeemsToBeInvalid[IndexRequest]("Index name is invalid")
+      }
+    Set(indexName)
+  }
+
+  private def update(request: GetRequest,
+                     indices: NonEmptyList[IndexName],
+                     filter: Option[Filter]): ModificationResult = {
+    val indexName = indices.head
+    request.index(indexName.value.value)
+    ModificationResult.UpdateResponse(filterResponse(filter, indexName, request.id()))
+  }
+
+  private def filterResponse(filter: Option[Filter],
+                             indexName: IndexName,
+                             documentId: String)
+                            (actionResponse: ActionResponse): ActionResponse = {
+    actionResponse match {
+      case response: GetResponse =>
+        filter match {
+          case Some(definedFilter) if response.isExists =>
+            verifyDocumentAccessibility(response, definedFilter) match {
+              case Inaccessible => GetApi.doesNotExistResponse(response)
+              case Accessible => response
+            }
+          case _ => response
+        }
+      case other => other
+    }
+  }
+
+  private def verifyDocumentAccessibility(originalResponse: GetResponse,
+                                          definedFilter: Filter) = {
+    val searchRequest = createSearchRequest(nodeClient, definedFilter)(originalResponse.asDocumentWithIndex)
+    Try(searchRequest.get()) match {
```

Comment:
note that this is synchronous request. IMO current thread will be blocked. We should use async version.